ngraph_bridge/ngraph_pipelined_tensors.cc

```diff
@@ -91,30 +91,23 @@ bool IndexLibrary::is_free(size_t id) {
 
 PipelinedTensorsStore::PipelinedTensorsStore(PipelinedTensorMatrix in,
                                              PipelinedTensorMatrix out)
-    : m_in_tensors(in),
-      m_out_tensors(out),
-      m_num_inputs(in.size()),
-      m_num_outputs(out.size()) {
+    : m_in_tensors(in), m_out_tensors(out) {
+  auto m_depth_in = in.size();
+  auto m_depth_out = out.size();
+
+  // We assume that input and output depths are same
+  m_depth = std::min(m_depth_in, m_depth_out);
+
   // The executable could have no inputs or no outputs.
   // Hence the if-else below to determine m_depth
-  bool has_inputs = in.size() > 0;
-  bool has_outputs = out.size() > 0;
-  if (has_inputs) {
-    if (has_outputs) {
-      auto m_depth_in = in[0].size();
-      auto m_depth_out = out[0].size();
-      // We assume that input and output depths are same
-      m_depth = std::min(m_depth_in, m_depth_out);
-    } else {
-      m_depth = in[0].size();
-    }
-  } else {
-    if (has_outputs) {
-      m_depth = out[0].size();
-    } else {
-      m_depth = 0;  // The executable has no inputs and outputs
-    }
+  bool has_inputs = in[0].size() > 0;
+  bool has_outputs = out[0].size() > 0;
+  if (has_inputs && !has_outputs) {
+    m_depth = in.size();
+  } else if (!has_inputs && has_outputs) {
+    m_depth = out.size();
```

Comment:
Maybe we dont need this if-else here. We can just check `m_depth_in== m_depth_out`. and pick that as m_depth